crates/amaru-ledger/src/rules/transaction/output_size.rs

```diff
@@ -0,0 +1,53 @@
+use amaru_kernel::{
+    cbor, protocol_parameters::ProtocolParameters, TransactionBody, TransactionOutput,
+};
+
+use crate::rules::TransactionRuleViolation;
+
+pub struct OutputTooSmall {
+    pub outputs_too_small: Vec<TransactionOutput>,
+}
+
+impl Into<TransactionRuleViolation> for OutputTooSmall {
+    fn into(self) -> TransactionRuleViolation {
+        TransactionRuleViolation::OutputTooSmall(self)
+    }
+}
+
+pub fn validate_output_size(
+    transaction: &TransactionBody,
+    protocol_parameters: &ProtocolParameters,
+) -> Result<(), OutputTooSmall> {
+    let coins_per_utxo_byte = protocol_parameters.coins_per_utxo_byte;
+    let outputs_too_small = transaction
+        .clone()
```

Comment:
Hugh. If anything, you should clone the `outputs`,  (`.iter().cloned()`) but not the entire transaction. Still, I don't think validation rules should be cloning anything. 

Rather, we should introduce a lifetime on the error which is bound to the transaction body.

Reply:
Yeah, this is something I feel is pretty weak in my Rust abilities -- avoiding cloning with better, and perhaps more complicated, idiomatic solutions. I'll add a TODO to clean up these clones.

Reply:
Is there any ways we can help with that? Perhaps we can do some pairing to _remove those clone_ and see where this leads us?

Reply:
Yeah, that could absolutely be helpful. I can also work with Pi or another Sundae person on that, to avoid taking you off your tasks.